homeassistant/components/ecovacs/vacuum.py

```diff
@@ -95,7 +97,16 @@ def __init__(self, device: sucks.VacBot) -> None:
 
         self.error: str | None = None
         self._attr_unique_id = vacuum["did"]
-        self._attr_name = vacuum.get("nick", vacuum["did"])
+
+        if (name := vacuum.get("nick")) is None:
+            name = vacuum.get("name", vacuum["did"])
```

Comment:
Why don't you chain them? Vacuum.get(.., vacuum.get(.., ..))

Reply:
because `nick` could als be present and have the value `None` (_checked with my own bot_) those it would already stopp on first get in the chain and return `None`